src/System.Data.SqlClient/src/System/Data/SqlClient/SqlConnection.cs

```diff
@@ -471,7 +471,7 @@ override public void Close()
             // connection is already in a closed state. this doesn't seem to be a 
             // problem except for logging, as we'll get duplicate Before/After/Error
             // log entries
-            if (previousState == ConnectionState.Open)
+            if (previousState != ConnectionState.Closed)
```

Comment:
@saurabh500 As per our discussion, I changed the logging to now log for non-closed connections instead of just open connections.  Let me know if this is what you had in mind or if you want to see any changes. // @avanderhoorn